### PR TITLE
Added support for yfinance api and updated requirements

### PR DESCRIPTION
**Pull request by t0tl**

### CODE REVIEW

**Comment by t0tl** [comment]
Add mocking for the yfinance API

File: portfolio_sim/utils.py [hunk 1]
```diff
@@ -110,3 +119,27 @@ def get_company_data(self, comp_name):
...
+    def __init__(self):
+        self.session = CachedLimiterSession(
+            per_second=0.9,
+            bucket_class=MemoryQueueBucket,
+            backend=SQLiteCache("yfinance.cache"),
+        )
+        return
+
+    def get_company_data(self, comp_name):
+        # TODO:
```
Comment (MMorsell) [inline portfolio_sim/utils.py:135]: Usually good to add your name or initials to a TODO, such as todo(mm):

File: portfolio_sim/utils.py [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
+from requests_cache import CacheMixin, SQLiteCache
+from requests_ratelimiter import LimiterMixin, MemoryQueueBucket
+import numpy as np
+
+
+class CachedLimiterSession(CacheMixin, LimiterMixin, requests.Session):
+    """A class for caching and limiting requests"""
 
 
 class APIRetriever(ABC):
```
Comment (MMorsell) [inline portfolio_sim/utils.py:15]: should the APIReciever logic along with the implementations all be in one file called `utils.py`? Maybe it make sense to extract this logic to another file with a better name

**Review by MMorsell — CHANGES_REQUESTED** [review]
@t0tl Added some thoughts

**Comment by t0tl** [comment]
> @t0tl Added some thoughts

Good comments, added some changes which build on your suggestions

**Pull request merged**